File: Python/Easy/2946 Matrix Similarity After Cyclic Shifts.py
class Solution:
    def areSimilar(self, mat: List[List[int]], k: int) -> bool:
        n = len(mat[0])
        # After k shifts, the row must be equal to its original state.
        # This is only true if every element matches the element k positions away.
        for row in mat:
            for j in range(n):
                if row[j] != row[(j + k) % n]:
                    return False
        return True

File: Java/Easy/2946 Matrix Similarity After Cyclic Shifts.java
class Solution {
    public boolean areSimilar(int[][] mat, int k) {
        int m = mat.length;
        int n = mat[0].length;
        k %= n; // Optimization: k shifts is the same as k % n shifts
        
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                // Check if the element at j matches the element at its shifted position
                if (mat[i][j] != mat[i][(j + k) % n]) {
                    return false;
                }
            }
        }
        return true;
    }
}
